pkg/controller/machine_safety.go

```diff
@@ -0,0 +1,482 @@
+		c.updateTimeStamp(newMS)
+	}
+}
+
+// addMachineToSafety enqueues into machineSafetyQueue when a new machine is added
+func (c *controller) addMachineToSafety(obj interface{}) {
+	machine := obj.(*v1alpha1.Machine)
+	c.enqueueMachineSafetyKey(machine)
+}
+
+// updateTimeStamp adds an annotation indicating the last time the number of replicas
+// of machineSet was changed
+func (c *controller) updateTimeStamp(ms *v1alpha1.MachineSet) {
+	for {
+		// Get the latest version of the machineSet so that we can avoid conflicts
+		ms, err := c.controlMachineClient.MachineSets(ms.Namespace).Get(ms.Name, metav1.GetOptions{})
+		if err != nil {
+			// Some error occued while fetching object from API server
+			glog.Error(err)
+			break
+		}
+		clone := ms.DeepCopy()
+		if clone.Annotations == nil {
+			clone.Annotations = make(map[string]string)
+		}
+		clone.Annotations[LastReplicaUpdate] = metav1.Now().Format("2006-01-02 15:04:05 MST")
+		_, err = c.controlMachineClient.MachineSets(clone.Namespace).Update(clone)
+		if err == nil {
+			break
+		}
+		// Keep retrying until update goes through
+		glog.Warning("Updated failed, retrying")
+	}
+}
+
+// enqueueMachineSafetyKey enqueues into machineSafetyQueue
+func (c *controller) enqueueMachineSafetyKey(obj interface{}) {
+	c.machineSafetyQueue.Add("")
+}
+
+// deleteOrphanVM teriminates's the VM on the cloud provider passed to it
+func (c *controller) deleteOrphanVM(vm driver.VM, secretRef *corev1.Secret, kind string, machineClass interface{}) {
+
+	glog.V(2).Infof("Orphan VM found. Initializing termination on VM: %s, %s", vm.MachineName, vm.MachineID)
+
+	dvr := driver.NewDriver(
+		vm.MachineID,
+		secretRef,
+		kind,
+		machineClass,
+		vm.MachineName,
+	)
+
+	err := dvr.Delete()
+	if err != nil {
+		return
+	}
+}
+
+// freezeOvershootingMachineObjects freezes machineSets and machineDeployment (who is the owner of the machineSet)
+func (c *controller) freezeOvershootingMachineObjects(machineSet *v1alpha1.MachineSet, reason string, message string) {
+
+	glog.V(2).Infof("Freezing MachineSet %q due to %q", machineSet.Name, reason)
+
+	for {
+		// Get the latest version of the machineSet so that we can avoid conflicts
+		machineSet, err := c.controlMachineClient.MachineSets(machineSet.Namespace).Get(machineSet.Name, metav1.GetOptions{})
+		if err != nil {
+			// Some error occued while fetching object from API server
+			break
+		}
+		clone := machineSet.DeepCopy()
+		newStatus := clone.Status
+		mscond := NewMachineSetCondition(v1alpha1.MachineSetFrozen, v1alpha1.ConditionTrue, reason, message)
+		SetCondition(&newStatus, mscond)
+		if clone.Labels == nil {
+			clone.Labels = make(map[string]string)
+		}
+		clone.Status = newStatus
+		clone.Labels["freeze"] = "True"
+		_, err = c.controlMachineClient.MachineSets(clone.Namespace).Update(clone)
+		if err == nil {
+			break
+		}
+		// Keep retrying until update goes through
+		glog.Warning("Updated failed, retrying")
```

Comment:
Elaborate the comment with error message?

Reply:
Done. Added error message print.